WinImageBuilder.psm1

```diff
@@ -682,7 +685,10 @@ function Get-VirtIODrivers {
     foreach ($driver in $VirtioDrivers) {
         foreach ($osVersion in $VirtIODriverMappings.Keys) {
             $map = $VirtIODriverMappings[$osVersion]
-            if (!(($map[0] -eq $MajorMinorVersion) -and ($map[1] -eq $isServer))) {
+            if (!(($map[0] -eq $MajorMinorVersion) -and ($map[2] -eq $isServer))) {

```

Comment:
here you are required to do a greater comparison, as 2016 or 2018 have more than one build number, it s rather a build number interval.

Reply:
https://docs.microsoft.com/en-us/windows-server/get-started/windows-server-release-info

Reply:
Hmm, I don't think so, build numbers of **compatible** windows versions don't change. If the build number changes, then the driver might crash anyway. So it is safer to just check for equality.
In your link, you can see that easily: Windows Server 2019 and Windows Server version 1809 (the 2nd and 3rd Entry) have exactly the same build number.
People in [this](https://techcommunity.microsoft.com/t5/Windows-Server-Insiders/Windows-Server-2019-version-info/td-p/234472) discussion came to the same conclusion.

Reply:
this way, you might end up not installing the virtio drivers for some windows versions. We can look at also the Get-ItemProperty -Path "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion" -Name ReleaseId. As this tool should support intermediary (build) windows versions (test builds, etc), it is mandatory to check for the interval.

Reply:
I have checked the 0.1.171 and the files for 2k19 have the same hash as the files for 2k16. By default, the imaging tool code should install the 2k16 drivers for 2k19, which are the same. For the moment, there is no need for this change and I will come back with a refactored code when the stable virtio drivers contain different drivers for 2k19.